pgmult/lda.py

```diff
@@ -391,9 +413,20 @@ def __init__(self, data, timestamps, K, alpha_theta):
     def initialize_parameters(self):
         self.sigmasq_states = 0.1  # TODO make this learned, init from hypers
 
-        mean_psi = compute_uniform_mean_psi(self.V)[0][None,:,None]
-        self.psi = np.tile(mean_psi, (self.T, 1, self.K))
-
+        # finale: added this to the initialization (must already be
+        # correct size, do that in the top level file or as another
+        # function in the top level file), not here
+        #
+        # Melih: pi_to_psi expects rows to be normalized
+        # and only takes arrays of size <= 2.
+        # Need confirmation on whether double transpose below
+        # results in desired behavior.
+        if self.param_set['use_init_beta']:
+            self.psi = np.array(map(lambda x: pi_to_psi(x.T).T, self.param_set['init_beta']))
```

Comment:
Great to see that it's doing better with the initialization!  Let's chat
about this with Matt and Scott tomorrow (maybe leave a longer run going to
see whether LDA ever catches up if given another 50 or so iterations --
hopefully not)

On Tue, Jul 28, 2015 at 10:52 AM, Melih Elibol <notifications@github.com>
wrote:

> In pgmult/lda.py
> <https://github.com/dtak/pgmult/pull/1#discussion_r35656341>:
>
> > @@ -391,9 +413,20 @@ def __init__(self, data, timestamps, K, alpha_theta):
> >      def initialize_parameters(self):
> >          self.sigmasq_states = 0.1  # TODO make this learned, init from hypers
> >
> > -        mean_psi = compute_uniform_mean_psi(self.V)[0][None,:,None]
> > -        self.psi = np.tile(mean_psi, (self.T, 1, self.K))
> > -
> > +        # finale: added this to the initialization (must already be
> > +        # correct size, do that in the top level file or as another
> > +        # function in the top level file), not here
> > +        #
> > +        # Melih: pi_to_psi expects rows to be normalized
> > +        # and only takes arrays of size <= 2.
> > +        # Need confirmation on whether double transpose below
> > +        # results in desired behavior.
> > +        if self.param_set['use_init_beta']:
> > +            self.psi = np.array(map(lambda x: pi_to_psi(x.T).T, self.param_set['init_beta']))
>
> @finaledoshivelez <https://github.com/finaledoshivelez> modified
> assignment to psi. Results:
> [image: dtm_from_lda]
> <https://cloud.githubusercontent.com/assets/1884921/8934613/804e7dd6-3516-11e5-98cf-ad91933d8467.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dtak/pgmult/pull/1/files#r35656341>.
>